sys/net/application_layer/gcoap/fileserver.c

```diff
@@ -275,7 +275,13 @@ static ssize_t _put_file(coap_pkt_t *pdu, uint8_t *buf, size_t len,
         if ((ret = vfs_lseek(fd, 0, SEEK_END)) < 0) {
             goto close_on_error;
         }
-        if (block1.offset != (unsigned)ret) {
+        if (block1.offset < (unsigned)ret) {
```

Comment:
Oh this could actually happen, I guess.
A very strict CoAP implementation would maybe ignore the message with a message ID because it is reused to early by some endpoint. But I think not checking for this is a good tradeoff between correctness and performance.
Have you experienced this case?

Reply:
Yes this happens with the added test.
The links are set to have 25% packet loss, so an ACK being lost and the block being re-transmitted is something that happens there often.

Reply:
Ok, good to have this..